### PR TITLE
Sdk examples

### LINKED ISSUE #660: Port rasa core examples to SDK code
Make sure all examples that are part of rasa core work with the sdk code. 

Please also update the documentation of the examples to reflect any changes.

- [x] moodbot
- [x] restaurantbot
- [x] concertbot (needs actions endpoint)
- [x] remotebot
- [x] interactive learning examples

List of doc changes:
 - [ ] connectors (still a WIP)
 - [x] custom actions - this might be a good place to mention how to run the actions server
 - [x] interactive learning
 - [x] domains - custom actions section
 - [x] agent (outdated functions)
 - [ ] migration guide

### PR DESCRIPTION
**Pull request by akelad**
**Proposed changes**:
- update examples and their docs to the SDK code
- fixes #660 

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [ ] added some tests for the functionality
- [x] updated the documentation
- [ ] updated the changelog

### CODE REVIEW

File: docs/quickstart.rst [hunk 1]
```diff
@@ -190,27 +190,25 @@ into ``models/dialogue``.
...
 If you are running the cells here in the docs, run this cell:
 
 .. runnable::
 
-   from rasa_core.agent import Agent
-   from rasa_core.channels.console import ConsoleInputChannel
-   agent = Agent.load('models/dialogue')
-   agent.handle_channel(ConsoleInputChannel())
+   from rasa_core.run import start_server
+   start_server('models/dialogue')
```
Comment (akelad) [inline docs/quickstart.rst]: not entirely sure this is going to work from the docs, but couldn't test it either

Comment (tmbo) [reply docs/quickstart.rst]: @amn41 do you know if we can start an http server here?

Comment (amn41) [reply docs/quickstart.rst]: yes you can! check out step 5 in http://rasa.com/docs/nlu/quickstart/

Comment (tmbo) [reply docs/quickstart.rst]: nice alright, then we'll keep it and try it out :rocket:

**Pull request merged**